manifests/config.pp

```diff
@@ -59,6 +60,43 @@
 
     }
 
+    'Suse': {
+      $pamconfig_mkhomedir_check_cmd = '/usr/sbin/pam-config -q --mkhomedir | grep session:'
+      $pamconfig_check_cmd           = '/usr/sbin/pam-config -q --sss | grep session:'
+
+      if $mkhomedir {
+
+        exec { 'pam-config -a --mkhomedir':
+          path   => '/bin:/usr/bin:/sbin:/usr/sbin',
+          unless => $pamconfig_mkhomedir_check_cmd,
+        }
+
+      }
+
+      exec { 'pam-config -a --sss':
+        path   => '/bin:/usr/bin:/sbin:/usr/sbin',
+        unless => $pamconfig_check_cmd,
+      }
+
+      if $manage_nsswitch {
+
```

Comment:
I would suggest replacing the following two execs with the aforementioned nsswitch module and moving it out of the Suse block.  Others may wish to use this on all supported OSes and the nsswitch module supports them.

We also need some spec tests for turning on and off the $manage_nsswitch param.
Let me know if you need help writing tests... Glad to help ;)

```puppet
if $manage_nsswitch {
  include '::nsswitch::params'

  class { 'nsswitch':
    passwd  => concat($::nsswitch::params::passwd_default, 'sss'),
    shadow  => concat($::nsswitch::params::shadow_default, 'sss'),
    group   => concat($::nsswitch::params::group_default, 'sss'),
    sudoers => concat($::nsswitch::params::sudoers_default, 'sss'),
  }
}
```

https://github.com/trlinkin/puppet-nsswitch/blob/master/manifests/init.pp

This is exactly what I am doing in my profile, but I can't argue that it may be nice to let the sssd module do this, until you don't like its settings...
I suspect this is a good default set of items to add sssd to.